Libs/MRML/Core/vtkMRMLScene.cxx

```diff
@@ -1137,7 +1137,7 @@ vtkMRMLNode*  vtkMRMLScene::AddNodeNoNotify(vtkMRMLNode *n)
       // their references.
       std::string newId(sn->GetID());
       std::string oldId(n->GetID() ? n->GetID() : sn->GetID());
-      if (oldId != newId)
+      if (oldId != newId && !oldId.empty())
```

Comment:
oldId will never be empty. If sn->GetID() is NULL then these lines will crash:
std::string newId(sn->GetID()); 
std::string oldId(n->GetID() ? n->GetID() : sn->GetID());